src/app/App.tsx

```diff
@@ -1,10 +1,67 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { calculateNationalInsurance, NationalInsuranceDataType } from "./api";
+import { Button, NumberInput, SubTitle, Panel, Text } from "./components/atoms";
+import { Header, TaxDifference, TaxInformation } from "./components/molecules";
 
-const App = () => (<div>
-  <h1 className="text-slate-100">Hello</h1>
-</div>
+const App = () => {
+  useDispatch()
+  const [income, setIncome] = useState(0);
+  const [taxFrom2018, setTaxFrom2018] = useState<NationalInsuranceDataType>();
+  const [taxFrom2019, setTaxFrom2019] = useState<NationalInsuranceDataType>();
 
-);
+  const [calcTax2018, calcTax2019] = (['2018-04-06', '2019-04-06'] as const).map(d => calculateNationalInsurance(d))
```

Comment:
I have not worked much with Ramda before, but my understanding was that since it was curried and not completed, it would not have fired the network call. As I understand, `calcTax2018` should be like 
```
(income: number) => calculateNationalInsurance(taxYearFromMap, income)
```

Does this align with your understanding? I think it could benefit from a `useMemo` but I do not believe any network calls happen until the `handleSubmit`. Please do correct if this is not how Ramda handles curry

Also worth noting that I am not super fond of having the calls broken out like this either, I'd prefer something that perhaps had an array or map, but I was trying to squeeze in the curry(🙊)!